Document `RestClient.request`’s `version` param

lawrence-forooghian opened the pull request:
Based on specs at commit https://github.com/ably/specification/commit/505238c.

owenpearson reviewed (changes requested):
A couple of thoughts:
- I think we could do better than "protocol version" here. I don't expect anyone reading this to be certain of what it means and they'll likely end up at https://www.google.com/search?q=ably+protocol+version which won't help much either. Perhaps something like "major REST API version" would make it more obvious?
- "Any Ably documentation which recommends the use of this method will specify what value to pass for this parameter" really ought to be the case but that's certainly not the case now and it would be a significant task to update all the documentation which uses `Rest.request`. I'd be tempted to just link out to https://ably.com/docs/api/rest-api#versioning instead.

lawrence-forooghian commented:
Thanks for the feedback @owenpearson, I’ve changed it based on your feedback. The comment about "Any Ably documentation which recommends the use of this method will specify what value to pass for this parameter" was based on this comment in [ADR61](https://ably.atlassian.net/wiki/spaces/ENG/pages/2222456951/ADR61+Versioning+of+realtime+API+endpoints):

> In all documentation and casual interactions where we recommend to customers that they use rest#request, and tell them what they should expect, we should give them a protocol version to supply.

But from what you're saying, that's more of a that'd-be-great than a we're-planning-on-doing-this, so I’ve removed it.

m-hulbert commented on descriptions.md:
```diff
@@ -16,9 +16,10 @@ A client that offers a simple stateless API to interact directly with Ably's RES
+| request(String method, String path, Int version, `Dict<String, String>` params?, JsonObject \| JsonArray body?, `Dict<String, String>` headers?) => io HttpPaginatedResponse ||  | RSC19 | Makes a REST request to a provided path. This is provided as a convenience for developers who wish to use REST API functionality that is either not documented or is not yet included in the public API, without h ...
 || `method` ||| The request method to use, such as `GET`, `POST`. |
 || `path` ||| The request path. |
+|| `version` || RSC19f1 | The major version of the Ably REST API to use. For more information about REST API versioning, see [this section of the REST API reference](https://ably.com/docs/api/rest-api#versioning). |
```
```suggestion
|| `version` || RSC19f1 | The major version of the Ably REST API to use. See the [REST API reference](https://ably.com/docs/api/rest-api#versioning) for information on versioning. |
```

m-hulbert reviewed (commented):
One minor suggestion just to keep the text consistent with the description for `params`.

m-hulbert reviewed (approved):
LGTM
